### DISCUSSION TITLE
Registration process

### DISCUSSION THREAD

**Discussion by rajeev-k-tomy** [hyva-themes/magento2-react-checkout#220] [General]
One of the most requested features is the customer registration feature within checkout. This is a place where to get the best customer registration flow.

The one in my mind is, provide a password/confirm-password field in the login form section and collect that information. After that, the user will proceed as a guest user does and on the success page, we will cover the guest to the user to a real customer using the provided password and attach the order to the newly created customer.

Or do we have more cleaner approach?

**Comment by jesse-deboer** [comment]
The following flow might also be an option:
- createEmptyCart
- createCustomer
- createEmptyCart
- mergeCarts

But I don't know if this is beneficial to the user because there will be a load time in the checkout. It might also be possible to do this during the placeOrder call. What do you think?

**Reply by rajeev-k-tomy** [reply]
@jesse-deboer Yes, the problem with the suggested approach is already pointed out by you. It is not at all good to give such a load time in the checkout. Instead, we need to do it after placing the order. Yes, this is what I am also thinking about.

**Reply by EmperorKonstantin** [reply]
Well how does PWA Studio handle it?
